src/components/common/BlockExplorerLink.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { ReactElement } from 'react'
+import { useWalletConnection } from 'hooks/useWalletConnection'
+import { abbreviateString } from 'utils'
+import { Network } from 'types'
+
+type BlockExplorerLinkType = 'tx' | 'address' | 'contract' | 'token' | 'event'
+
+export interface Props {
+  /**
+   * type of BlockExplorerLink
+   */
+  type: BlockExplorerLinkType
+  /**
+   * address or transaction or other hash
+   */
+  identifier: string
+  /**
+   * network number | chain id
+   */
+  networkId?: number
+  /**
+   * label to replace textContent generated from indentifier
+   */
+  label?: string | ReactElement | void
+
+  /**
+   * Use the URL as a label
+   */
+  useUrlAsLabel?: boolean
+  /**
+   * className to pass on to <a/>
+   */
+  className?: string // to allow subclassing styles with styled-components
+}
+
+function getEtherscanUrlPrefix(networkId: Network): string {
+  return !networkId || networkId === Network.Mainnet ? '' : (Network[networkId] || '').toLowerCase() + '.'
+}
+
+function getEtherscanUrlSuffix(type: BlockExplorerLinkType, identifier: string): string {
+  switch (type) {
```

Comment:
Does TS not complain about no `default`?

Reply:
No, that's something I liked in this case about TS

It knows all possible types, and it knows I'm being exhaustive, so it knows I always return a string
If I delete any of the cases, then he complains

Reply:
TS smart :mortar_board: